test/e2e/common_suite_test.go

```diff
@@ -870,3 +874,253 @@ func doTestCreatePeerPodWithAuthenticatedImageWithoutCredentials(t *testing.T, a
 	pod := newPod(namespace, podName, podName, imageName, withRestartPolicy(v1.RestartPolicyNever))
 	newTestCase(t, "InvalidAuthImagePeerPod", assert, "Peer pod with Authenticated Image with Invalid Credentials has been created").withPod(pod).withAuthenticatedImage().withAuthImageStatus(expectedAuthStatus).withCustomPodState(v1.PodPending).run()
 }
+
+func doTestCaaDaemonsetUpgrade(t *testing.T) {
+	runtimeClassName := "kata-remote"
+	namespace := envconf.RandomName("default", 7)
+	deploymentName := "nginx-deployment"
+	containerName := "nginx"
+	imageName := "nginx"
+	serviceName := "nginx-service"
+	portName := "port80"
+	rc := int32(2)
+	labelsMap := map[string]string{
+		"app": "nginx",
+	}
+	verifyPodName := "verify-pod"
+	verifyContainerName := "verify-container"
+	verifyImageName := "radial/busyboxplus:curl"
+
+	deployment := &appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      deploymentName,
+			Namespace: namespace,
+			Labels:    labelsMap,
+		},
+		Spec: appsv1.DeploymentSpec{
+			Replicas: &rc,
+			Selector: &metav1.LabelSelector{
```

Comment:
How did we make sure each nginx pods assigned to different workers?

Reply:
I did nothing. But I didn't noticed the 2 pods would be deployed on 1 worker if there are 2 schedulable workers

Reply:
@genjuro214 it's not always the case. It's possible the 2 pods assigned on same worker, I met it before.